### PR TITLE
Add AWS Devicefarm job back into CI

### PR DESCRIPTION
**Pull request by sergeichestakov**
# Why

We want test-suite running all the tests on a device in devicefarm after every commit. This PR adds that functionality back into CI.

# How

I added the CI job to run test-suite on AWS back into the `.circleci/config.yml` within the client workflow. I also switched to publishing the tests on a channel based off of the commit SHA rather than the branch name so that we can have separate published versions on the same branch for each new commit. I also updated some of the tests that have been failing, starting with `Basic.js`.

# Test Plan

This should run as a CI job in the new `.circleci/config.yml`. It will fail during the client workflow initially as 27 of the tests do not currently pass. This PR hopes to address that and get it back to green by either fixing the tests or commenting them/assigning them to others for now until we can get them fixed.

### CODE REVIEW

**Review by esamelson — COMMENTED** [review]
this looks good so far but let's hold off on merging until we can get it green (or have a clear plan/path to do so).

File: packages/expo-av/android/src/main/java/expo/modules/av/AVManager.java [hunk 1]
```diff
@@ -500,7 +500,7 @@ public void runWithVideoView(final VideoView videoView) {
   // Recording API
 
   private boolean isMissingAudioRecordingPermissions() {
-    return mModuleRegistry.getModule(Permissions.class).getPermission(Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_GRANTED;
+    return mModuleRegistry.getModule(Permissions.class).getPermission(Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED;
```
Comment (sjchmiela) [inline packages/expo-av/android/src/main/java/expo/modules/av/AVManager.java]: 😅 😅 😅

**Comment by sergeichestakov** [comment]
Ok after fixing and removing the tests that were failing, all of the tests that run in CI now pass so the client workflow is all green 👍

**Review by esamelson — APPROVED** [review]
this looks good to me! you can go ahead and merge 🚀

**Pull request merged**